webapp/channels/src/components/widgets/modals/components/react_select_item.tsx

```diff
@@ -30,13 +31,33 @@ type Props = BaseSettingItemProps & {
     handleChange: (selected: OnChangeValue<Option, boolean>) => void;
 }
 
+// Function to extract text from FormattedMessage components
+export const getOptionLabel = (option: Option, intl: ReturnType<typeof useIntl>) => {
```

Comment:
Ideally, we have a shared component for select that has this built it for all usages of ReactSelect. But given that we are currently reusing directly everywhere, I exported this here and used in other places. Open to suggestions if we don't want to export from `widgets/modals/components...`

Reply:
Yeah, I agree that a common component would be better, but I think having a reusable function for this is a decent alternative